snuba/datasets/schemas/join.py

```diff
@@ -40,7 +41,7 @@ def __str__(self) -> str:
             f"{self.right.table_alias}.{self.right.column}"
 
 
-class JoinedSource(ABC):
+class JoinedSource(RelationalSource, ABC):
```

Comment:
I think there will be some refactoring in this file overall to make the relationship between these classes and RelationalSource more obvious